lib/active_storage/send_zip_helper.rb

```diff
@@ -81,7 +81,16 @@ def self.save_file_on_server(file, folder, subfolder: nil)
         filepath = File.join folder, filename
       end
 
-      File.open(filepath, 'wb') { |f| f.write(file.download) }
+      unless resize_to_limit.nil?
+        file = file.variant(resize_to_limit: resize_to_limit).processed
+      end
+
+      # read file from service
+      # contents = URI.open(file.service.send(:url_for, file.key)) { |f| f.read }
+
+      # write the file on disk
+      File.open(filepath, 'wb') { |f| f.write(file.service.download(file.key)) }
```

Comment:
Is there any difference between `file.download` & `file.service.download(file.key)` ?

Reply:
@madeindjs #download is not available on ActiveStorage::Variant, however Service#download is available on both.